src/main/java/com/googlecode/download/maven/plugin/internal/WGet.java

```diff
@@ -374,7 +375,11 @@ public void execute() throws MojoExecutionException, MojoFailureException {
     private void unpack(File outputFile) throws NoSuchArchiverException {
         UnArchiver unarchiver = this.archiverManager.getUnArchiver(outputFile);
         unarchiver.setSourceFile(outputFile);
-        unarchiver.setDestDirectory(this.outputDirectory);
+        if (unarchiver instanceof GZipUnArchiver) {
+            unarchiver.setDestFile(new File(this.outputDirectory, this.outputFileName.replaceFirst("(.+)\\.gz(?:ip)?$", "$1")));
```

Comment:
Somthing like `outputFileName.substring(0, outputFileName.lastIndexOf('.'))` should work in the general case.

Reply:
Okay. I will update the PR :-)